packages/x-charts/src/LineChart/AnimatedLine.tsx

```diff
@@ -40,6 +40,7 @@ const AnimatedLine = React.forwardRef<SVGPathElement, AnimatedLineProps>(
           fill="none"
           filter={ownerState.isHighlighted ? 'brightness(120%)' : undefined}
           opacity={ownerState.isFaded ? 0.3 : 1}
+          data-series={ownerState.id}
```

Comment:
Moved the `data-series` to the `AnimatedLine`/`AnimatedArea` components. IMO it makes more sense. The series ID is already provided through the `ownerState`. If someone is using their own slots, they might not need `data-series` to be set. If they want to, they can easily add `data-series={ownerState.id}` here.